### PR TITLE
Upgrade to PHP 8.3

### PR DESCRIPTION
**Pull request by loicsapone**
**This PR needs a good review before being merged!**

@julienj I upgrade required PHP version to 8.3 in order to later build binaries with frankenphp, so we don't need to support multiple PHP versions
This is the first (big) step in the upgrade work, the next step is to upgrade dependencies to Symfony 6.4 (or maybe 7.0) in a next PR
I also removed self update command because of the abandoned dependency and of future builds with frankenphp

### CODE REVIEW

**Comment by loicsapone** [comment]
Ah, Scrutinizer failed because of the PHP version used
Could you update it in Scrutinizer config @julienj?

**Comment by julienj** [comment]
@loicsapone I deleted Scrutinizer, it is no longer useful with cs-fixer, phpstan and rector.
Another commit is needed to validate that the deletion is effective

**Comment by loicsapone** [comment]
@julienj done

**Comment by julienj** [comment]
I did some real tests, it seems all good!

Little question : Why upgrade to phpunit 9 and not 11?

File: src/Automate/Plugin/AbstractNotificationPlugin.php [hunk 1]
```diff
@@ -15,61 +15,44 @@
...
-     * {@inheritdoc}
-     */
-    public static function getSubscribedEvents()
+    
+    abstract public function getName(): string;
+
+    abstract protected function sendMessage(string $message, string $eventName);
+
+    
+    abstract public function getConfigurationNode(): \Symfony\Component\Config\Definition\Builder\NodeDefinition;
```
Comment (julienj) [inline src/Automate/Plugin/AbstractNotificationPlugin.php:52]: use import instead

File: src/Automate/Plugin/CacheToolPlugin.php [hunk 2]
```diff
@@ -79,10 +70,8 @@ public function onTerminate(DeployEvent $event)
         }
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function getConfigurationNode()
+    
+    public function getConfigurationNode(): \Symfony\Component\Config\Definition\Builder\NodeDefinition
```
Comment (julienj) [inline src/Automate/Plugin/CacheToolPlugin.php:74]: use import instead

File: src/Automate/Plugin/GitlabPlugin.php [hunk 3]
```diff
@@ -135,10 +120,8 @@ private function send(ContextInterface $context, $message, $envName)
         }
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function getConfigurationNode()
+    
+    public function getConfigurationNode(): \Symfony\Component\Config\Definition\Builder\NodeDefinition
```
Comment (julienj) [inline src/Automate/Plugin/GitlabPlugin.php:124]: use import instead

File: src/Automate/Plugin/GitlabPlugin.php [hunk 4]
```diff
@@ -82,9 +73,9 @@ public function register(Project $project)
     /**
      * Create a successed JOB on Gitlab.
      */
-    public function onSuccess(DeployEvent $event)
+    public function onSuccess(DeployEvent $event): void
     {
-        if ($this->project) {
+        if ($this->project instanceof \Automate\Model\Project) {
```
Comment (julienj) [inline src/Automate/Plugin/GitlabPlugin.php:78]: use import instead

File: src/Automate/Plugin/GitlabPlugin.php [hunk 5]
```diff
@@ -94,23 +85,17 @@ public function onSuccess(DeployEvent $event)
     /**
      * Create a failed JOB on Gitlab.
      */
-    public function onFailed(FailedDeployEvent $event)
+    public function onFailed(FailedDeployEvent $event): void
     {
-        if ($this->project) {
+        if ($this->project instanceof \Automate\Model\Project) {
```
Comment (julienj) [inline src/Automate/Plugin/GitlabPlugin.php:90]: use import instead

File: src/Automate/Plugin/GitterPlugin.php [hunk 6]
```diff
@@ -14,18 +14,14 @@
...
     {
         return 'gitter';
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function getConfigurationNode()
+    
+    public function getConfigurationNode(): \Symfony\Component\Config\Definition\Builder\NodeDefinition
```
Comment (julienj) [inline src/Automate/Plugin/GitterPlugin.php:24]: use import instead

File: src/Automate/Plugin/SentryPlugin.php [hunk 7]
```diff
@@ -24,18 +24,14 @@
...
     {
         return 'sentry';
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function getConfigurationNode()
+    
+    public function getConfigurationNode(): \Symfony\Component\Config\Definition\Builder\NodeDefinition
```
Comment (julienj) [inline src/Automate/Plugin/SentryPlugin.php:34]: use import instead

File: src/Automate/Plugin/SlackPlugin.php [hunk 8]
```diff
@@ -25,20 +25,15 @@
...
     {
         return 'slack';
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function getConfigurationNode()
+    
+    public function getConfigurationNode(): \Symfony\Component\Config\Definition\Builder\NodeDefinition
```
Comment (julienj) [inline src/Automate/Plugin/SlackPlugin.php:35]: use import instead

**Comment by julienj** [comment]
Thank you very much @loicsapone , just very small feedback on the imports, the rest is ok.

**Comment by loicsapone** [comment]
@julienj I fixed missing use statements

> Little question : Why upgrade to phpunit 9 and not 11?

Because I upgrade to minimum required version for PHP 8.3 in this PR.
PHPUnit 11 comes with Symfony 7.0 in another PR I prepared

**Pull request merged**